utils/src/main/java/io/questdb/cliutil/Table2IlpCopier.java

```diff
@@ -0,0 +1,238 @@
+
+
+                    try (Sender sender = buildLineTcpSender(params)) {
+                        String tableName = params.getDestinationTableName();
+
+                        try {
+                            while (resultSet.next()) {
+                                sendLine(symbols, resultSet, columnCount, columnNames, columnTypes, timestampIndex, sender, tableName);
+                                totalSentLines++;
+
+                                if (totalSentLines % 10000 == 0) {
+                                    long end = microsecondClock.getTicks();
+                                    long linesPerSec = 10000 * Timestamps.SECOND_MICROS / (end - start);
+                                    System.out.println(totalSentLines + " lines, " + linesPerSec + " lines/sec");
+                                    start = microsecondClock.getTicks();
+                                }
+
+                            }
+                        } catch (Exception th) {
+                            try {
+                                sender.flush();
+                            } catch (Exception flushError) {
+                                System.err.println("Failed to flush sender while handling error");
+                                flushError.printStackTrace();
+                            }
+                            throw th;
+                        }
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+            System.err.println("Failed to connect to the source");
+        } finally {
+            System.out.println("Total sent lines: " + totalSentLines);
+        }
+        return totalSentLines;
+    }
+
+    private static Sender buildLineTcpSender(Table2Ilp.Table2IlpParams params) {
+        Sender.LineSenderBuilder senderBuilder = Sender.builder();
+        senderBuilder.address(params.getDestinationIlpHost() + ":" + params.getDestinationIlpPort());
+        if (params.enableDestinationTls()) {
+            senderBuilder.enableTls();
+        }
+
+        if (params.getDestinationAuthKey() != null) {
+            senderBuilder
+                    .enableAuth(params.getDestinationAuthKey())
+                    .authToken(params.getDestinationAuthToken());
+        }
+        return senderBuilder.build();
+    }
+
+    private static LowerCaseCharSequenceHashSet createSymbolsSet(Table2Ilp.Table2IlpParams params) {
+        LowerCaseCharSequenceHashSet set = new LowerCaseCharSequenceHashSet();
+        String[] symbols = params.getSymbols();
+        //noinspection ForLoopReplaceableByForEach
+        for (int i = 0, n = symbols.length; i < n; i++) {
+            set.add(symbols[i]);
+        }
+        return set;
+    }
+
+    private static Connection getConnection(String connectionString) throws SQLException {
+        Properties properties = new Properties();
```

Comment:
Shouldn't user and password be command line arguments ?

Reply:
These are defaults, if you set them in jdbc connection string it will override the defaults.

Reply:
This is the example from readme to set login / password from command line

```bash
java -cp utils.jar io.questdb.cliutil.Table2Ilp -d trades -dc localhost:9009 -s "trades WHERE start_time in '2022-06'" \ 
     -sc "jdbc:postgresql://localhost:9812/qdb?user=account&password=secret&ssl=false" \
     -sym "ticker,exchagne" -sts start_time
```